examples/nextjs-blog/pages/blog/post.js

```diff
@@ -1,38 +1,89 @@
-import React from 'react'
+import React, { PureComponent } from 'react'
 import Markdown from 'markdown-to-jsx'
 import PropTypes from 'prop-types'
 import { flamelinkApp as app } from '../../utils/flamelink'
-import { getPostsWithMedia, getImageAlt } from '../../utils/post/post.util'
-
-const Post = function(props = {}) {
-  const { post } = props
-
-  if (post) {
-    const {
-      title,
-      date,
-      author: { displayName },
-      content,
-      imageURL,
-    } = post
-
-    return (
-      <div>
-        <h1>{title}</h1>
-        <h2>{displayName}</h2>
-        <p>{new Date(date).toLocaleDateString()}</p>
-        <Markdown>{content}</Markdown>
-        <img src={imageURL} alt={getImageAlt(post)} />
-      </div>
-    )
+import {
+  getPostsWithMedia,
+  getImageAlt,
+  getDateString,
+} from '../../utils/post/post.util'
+
+class Post extends PureComponent {
+  constructor(props) {
+    super(props)
+    this._subscription = null
+    this.state = {
+      post: null,
+    }
   }
 
-  return 'No post data!'
+  componentDidMount() {
+    // todo: this isn't so lekker because the path may change in future. is it
+    // possible to share the `query.slug` value from the server's execution with
+    // the client?
+    const { pathname } = location
+    const slug = pathname.replace('/blog/', '')
+
+    this._subscription = app.content.subscribe({
+      schemaKey: 'blogPost',
+      changeType: 'modified',
+      filters: [['slug', '==', slug]],
+      populate: true,
+      callback: async (error, response) => {
+        if (error) {
+          throw new Error(
+            'Something went wrong while retrieving all the content. Details:',
+            error
+          )
+        }
+
+        const [post] = await Promise.all(
+          getPostsWithMedia(Object.values(response || {}))
```

Comment:
The latest version of the SDK should include the `url` property for files when you `populate` them. I'll make sure the latest SDK is properly published today. I published last night, but something went wrong half-way through and only half of the packages got published.

Reply:
PR #8 should resolve this, so then you can remove the manual query